### PR TITLE
Feature/list targets

### PR DESCRIPTION
**Pull request by enzofab91**
**Trello reference:**
https://trello.com/c/Bir0tFHB

**Descripcion:**
This PR add support for get all the targets of the current logged user.

### CODE REVIEW

**Review by mrodriguez90 — APPROVED** [review]

File: db/migrate/20210422151926_create_targets.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+class CreateTargets < ActiveRecord::Migration[6.1]
+  def change
+    create_table :targets do |t|
+      t.string      :title,       null: false
+      t.float       :radius,      null: false
+      t.float       :latitude,    null: false
+      t.float       :longitude,   null: false
+      t.timestamps                null: false
+
+      t.references  :user,        null: false
```
Comment (sebastiancaraballo) [inline db/migrate/20210422151926_create_targets.rb:10]:
```suggestion
      t.references  :user, foreign_key: true, null: false
```
This adds a new annotation as well.

File: spec/requests/api/v1/sessions/destroy_spec.rb [hunk 2]
```diff
@@ -15,7 +15,12 @@
 
     it 'does not return a successful response' do
       subject
-      expect(response).to have_http_status(404)
+      expect(response).to have_http_status(:not_found)
```
Comment (sebastiancaraballo) [inline spec/requests/api/v1/sessions/destroy_spec.rb:18]:
```suggestion
      expect(response).to be_not_found
```

File: spec/requests/api/v1/targets/create_spec.rb [hunk 3]
```diff
@@ -0,0 +1,49 @@
+describe 'POST api/v1/targets', type: :request do
+  subject { post api_v1_targets_path, params: params, headers: auth_headers, as: :json }
+
+  describe 'POST create' do
+    let(:user)          { create :user }
+    let(:topic)         { create :topic }
+
+    let(:title)           { 'New target' }
+    let(:radius)          { 300 }
+    let(:coordinates)     { Hash(latitude: -34.6037389, longitude: -58.3837591) }
```
Comment (sebastiancaraballo) [inline spec/requests/api/v1/targets/create_spec.rb:10]:
```suggestion
    let(:coordinates)     { latitude: -34.6037389, longitude: -58.3837591 }
```

File: spec/requests/api/v1/targets/index_spec.rb [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    subject
+    expect(response).to be_successful
+  end
+
+  it 'returns user targets' do
+    subject
+    expect(json[:targets].pluck([:id])).to match_array(targets.pluck([:id]))
+  end
+
+  it 'does not returns other\'s users targets' do
```
Comment (sebastiancaraballo) [inline spec/requests/api/v1/targets/index_spec.rb:18]:
```suggestion
  it "does not return other's users targets" do
```

**Review by sebastiancaraballo — APPROVED** [review]
Good work 💪

**Pull request merged**